ssoPlugin/widget/index.html

```diff
@@ -0,0 +1,278 @@
+		// UTILITY METHODS
+		// ============================================================================================
+
+		/**
+		 * Runs the provided function when the document is fully loaded.
+		 *
+		 * @param {Function} fn The function to be run.
+		 */
+		const ready = (fn) => {
+			if (document.readyState != 'loading') fn()
+			else document.addEventListener('DOMContentLoaded', fn)
+		}
+
+		/**
+		 * Saves the provided data object to the browser's localStorage using the BuildFire library.
+		 *
+		 * @param   {Object}  data An object containing the data to save.
+		 * @returns {Promise}      Resolves with boolean true, rejects with an error message.
+		 */
+		const saveToLocalStorage = (data) => new Promise((resolve, reject) => {
+			buildfire.localStorage.setItem('@bif/sso', data, (err, result) => {
+				if (err || !result) reject('There was a problem saving to local storage.')
+				else resolve(true)
+			})
+		})
+
+		// ============================================================================================
+		// APPLICATION METHODS
+		// ============================================================================================
+
+		// Set DOM variables that we'll reference in the following methods thanks to lexical scoping.
+		const $body = document.querySelector('body')
+		const $alertMessage = document.getElementById('bif-sso__alert-message')
+		const $loginButton = document.getElementById('bif-sso__login-button')
+
+		/**
+		 * Marks the completion of initialization by changing some body classes.
+		 */
+		const initComplete = () => {
+			$body.classList.remove('uninitialized')
+			$body.classList.add('initialized')
+		}
+
+		/**
+		 * Updates the alert message placeholder with the provided message and type.
+		 *
+		 * Expected types are "error" or "success". Default is "notification" and has no special style.
+		 *
+		 * @param {String}  message      The message that should be displayed.
+		 * @param {String} messageType? The message type, from the list above.
+		 */
+		const alertMessage = (message, messageType = 'notification') => {
+			const filteredClassNames = $alertMessage.className
+				.split(' ')
+				.filter((className) => !className.startsWith('bif-sso__alert-message--'))
+			if (message) filteredClassNames.push(`bif-sso__alert-message--${messageType}`)
+			$alertMessage.className = filteredClassNames.join(' ').trim()
+			$alertMessage.innerHTML = message
+		}
+
+		/**
+		 * Validates that the required settings exist and aren't falsey.
+		 *
+		 * @param {Object} settings An object containing the settings.
+		 */
+		const areValidSettings = (settings) => {
+			return settings && settings.domain && settings.clientId
+		}
+
+		/**
+		 * Gets and validates the authentication API settings from the BuildFire datastore.
+		 *
+		 * @returns {Promise} Resolves with the settings object, rejects with an error message.
+		 */
+		const getAuthApiSettings = () => new Promise((resolve, reject) => {
```

Comment:
Wait, do we really need a named method for loading Auth Api Settings?

Reply:
Why not? The main purpose of this method is to wrap `buildfire.datastore.get` into a proper Promise.

Reply:
This was a troll comment in direct juxtaposition for my earlier suggestion to create a loadAuthApi utility method ;)

Reply:
:facepalm: